tpu_inference/distributed/tpu_connector_local.py

```diff
@@ -371,6 +442,152 @@ def get_finished(self,
         assert self.connector_worker is not None
         return self.connector_worker.get_finished()
 
+    def update_connector_output(self, connector_output: KVConnectorOutput):
+        assert self.connector_scheduler is not None
+        self.connector_scheduler.update_connector_output(connector_output)
+
+    def get_kv_connector_stats(self) -> KVConnectorStats | None:
+        if self.connector_worker is None:
+            return None
+        return self.connector_worker.get_kv_connector_stats()
+
+
+class _StagingBufferManager():
+    """ Bookkeeping the staging buffer inside the connector scheduler.
+    NOTE(jcgu): the operations (e.g., allocate, free, get) to staging buffer / blocks are NOT thread-safe.
+    But it's okay since there is only one connector scheduler instance.
+    """
+
+    def __init__(self, num_blocks: int):
+        self.num_blocks = num_blocks
+        # {req_id: list(num_occupied_staging_blocks)}
+        self._blocks_for_save: dict[ReqId, int] = {}
+        self._blocks_for_load: dict[ReqId, int] = {}
+
+        self._num_free_blocks: int = self.num_blocks
+        self._num_blocks_for_save: int = 0
```

Comment:
maybe put a comment that these two variables _num_blocks_for_load, _num_blocks_for_save track the actively used blocks across all requests for load and save respectively.